CMakeLists.txt

```
@@ -118,7 +118,7 @@ if(LIBUTP_BUILD_PROGRAMS AND NOT MSVC)
 
     target_compile_definitions(ucat
         PRIVATE
-            $<$<NOT:$<BOOL:${WIN32}>>:_POSIX_C_SOURCE=200112L>
+            $<$<NOT:$<BOOL:${WIN32}||${BSD}>>:_POSIX_C_SOURCE=200112L>
```

A couple of notes:
1. `BSD` is only defined since CMake 3.25 ([ref](https://cmake.org/cmake/help/latest/variable/BSD.html)) while this branch and main Transmission repo are using 3.12 ([ref](https://github.com/transmission/libutp/blob/post-3.4-transmission/CMakeLists.txt#L1), [ref](https://github.com/transmission/transmission/blob/main/CMakeLists.txt#L1)), need some sort or backward compatibility
2. The `||` syntax isn't supported by CMake and its generator expressions ([ref](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html)), and it also requires boolean values to be `0` or `1` (not `ON`/`TRUE` or `OFF`/`FALSE`, which `BSD` variable may contain), the actual expression will be a bit more lengthy than that, e.g.

```suggestion
            $<$<NOT:$<OR:$<BOOL:${WIN32}>,$<BOOL:${BSD}>>>:_POSIX_C_SOURCE=200112L>
```

---

Thank you, this isn't my bailiwick. Patch updated.